src/sctools/groups.py

```diff
@@ -0,0 +1,179 @@
+    Parameters
+    ----------
+    args:
+        file_names: array of files.the basename of inputs should be formated as 'samplename_qc'
+        output_name: prefix of output file name. the basename of outputs
+        includes the Picard metrics class name.
+    Returns
+    ----------
+        return: 0
+        return if the program completes successfully.
+    """
+    for file_name in file_names:
+        cell_id = os.path.basename(file_name).split('_qc')[0]
+        class_name = os.path.basename(file_name).split('.')[1]
+        parsed = picard.parse(file_name)
+        dat = pd.DataFrame.from_dict(parsed['metrics']['contents'])
+        dat.insert(0, 'Sample', cell_id)
+        dat.to_csv(output_name + "_" + class_name + '.csv', index=False)
+
+
+def write_aggregated_qc_metrics(file_names, output_name):
+    """Command line entrypoint to merge Picard metrics along with RSEM and HISAT2 log
+    Parameters
+    ----------
+    args:
+        file_names: array of files,such as Picard row metric, hisat2 metrics.
+        output_name: prefix of output file name.
+    Returns
+    ----------
+        return: 0
+        return if the program completes successfully.
+    """
+    df = pd.DataFrame()
+    for file_name in file_names:
+        dat = pd.read_csv(file_name, index_col=0)
+        print(dat.index)
+        df = pd.concat([df, dat], axis=1, join_axes=[dat.index])
+    df.to_csv(output_name + '.csv', index=True)
+
+
+def parse_hisat2_log(file_names, output_name):
+    """Command line entrypoint parse, aggreagete and write HISAT2 logs
+    Parameters
+    ----------
+    args:
+        file_names: array of HISAT2 log files. Basename of file indicates the alignment references
+        '_qc' indicates the genome reference and '_rsem' indicates the transcriptome reference alignment.
+        output_name: prefix of output file name. 
+    Returns
+    ----------
+        return: 0
+        return if the program completes successfully.
+    """
+    metrics = {}
+    tag = "NONE"
+    for file_name in file_names:
+        if '_qc' in file_name:
+            cell_id = os.path.basename(file_name).split('_qc')[0]
+            tag = "HISAT2G"
+        elif '_rsem' in file_name:
+            cell_id = os.path.basename(file_name).split('_rsem')[0]
+            tag = "HISAT2T"
+        with open(file_name) as f:
+            dat = f.readlines()
+            d = [x.strip().split(':') for x in dat]
+            # remove the first row of each section.
+            d.pop(0)
+            metrics[cell_id] = {x[0]: x[1].strip().split(' ')[0] for x in d}
+    df = pd.DataFrame.from_dict(metrics, orient='columns')
+    df.insert(0, "Class", tag)
+    df_T = df.T
+    df_T.to_csv(output_name + '.csv')
+
+
+def parse_rsem_cnt(file_names, output_name):
+    """Command line entrypoint parse, aggreagete and write RSEM cnt
+    Parameters
+    ----------
+    args:
+        file_names: array of RSEM cnt files. The basename of inputs should be '_rsem'
```

Comment:
basename is the full file name less the directory structure -- I think this refers to the suffix. However, I'd love to know what the base name needs to be, here.